goal_src/engine/nav/path.gc

```diff
@@ -5,3 +5,496 @@
 ;; name in dgo: path
 ;; dgos: GAME, ENGINE
 
+;; definition for method 9 of type path-control
+;; INFO: Return type mismatch int vs none.
+(defmethod dummy-9 path-control ((obj path-control))
+  (cond
+   ((nonzero? (logand (-> obj flags) (path-control-flag not-found)))
+    (when
+     (and
+      (type-type? (-> obj process type) process-drawable)
+      *display-entity-errors*
+      )
+     (let ((s5-0 add-debug-text-3d)
+           (s4-0 #t)
+           (s3-0 68)
+           )
+      (format
+       (clear *temp-string*)
+       "path data error in ~S"
+       (-> obj process name)
+       )
+      (s5-0
+       s4-0
+       (the-as bucket-id s3-0)
+       *temp-string*
+       (-> obj process root trans)
+       (new 'static 'rgba :r #x3)
+       (the-as vector2h #f)
+       )
+      )
+     )
+    )
+   ((let ((a0-5 obj))
+     (and
+      *display-path-marks*
+      (nonzero? (logand (-> a0-5 flags) (path-control-flag display)))
+      )
+     )
+    (dotimes (s5-1 (-> obj curve num-cverts))
+     (let ((s4-1 (&+ (-> obj curve cverts) (* s5-1 16))))
+      (if
+       (and
+        (nonzero? (logand (-> obj flags) (path-control-flag draw-line)))
+        (< s5-1 (+ (-> obj curve num-cverts) -1))
+        )
+       (add-debug-line
+        #t
+        (bucket-id debug-draw1)
+        (the-as vector s4-1)
+        (the-as vector (&+ (-> obj curve cverts) (* (+ s5-1 1) 16)))
+        (new 'static 'rgba :r #xff :g #x80 :a #x80)
+        #f
+        (the-as rgba -1)
+        )
+       )
+      (if (nonzero? (logand (-> obj flags) (path-control-flag draw-point)))
+       (add-debug-x
+        #t
+        (bucket-id debug-draw1)
+        (the-as vector s4-1)
+        (new 'static 'rgba :r #xff :a #x80)
+        )
+       )
+      (when (nonzero? (logand (-> obj flags) (path-control-flag draw-text)))
+       (let ((s3-1 add-debug-text-3d)
+             (s2-1 #t)
+             (s1-0 68)
+             )
+        (format (clear *temp-string*) "~D" s5-1)
+        (s3-1
+         s2-1
+         (the-as bucket-id s1-0)
+         *temp-string*
+         (the-as vector s4-1)
+         (new 'static 'rgba :r #x4)
+         (the-as vector2h #f)
+         )
+        )
+       )
+      )
+     )
+    )
+   )
+  0
+  (none)
+  )
+
+;; definition for method 16 of type path-control
+(defmethod path-distance path-control ((obj path-control))
+  (let ((f30-0 0.0))
+   (dotimes (s5-0 (+ (-> obj curve num-cverts) -1))
+    (+!
+     f30-0
+     (vector-vector-distance
+      (the-as vector (&+ (-> obj curve cverts) (* s5-0 16)))
```

Comment:
Can you try changing `cverts` to an `(inline-array vector)`?  I think this would clean up this stuff.